Implemented unit test for "Update Post" feature

Ufuk-Arslan opened the pull request:
I've added those changes:
- Changed the return value of **updatePost** function from PostEntitiy class to integer to provide consistency with the **createPost** function.
- Created **UpdatePostControllerTest** which creates a new post object, gets its id, then sends a mock update request to the post with that id.  
- Added **updatePost** function to **PostServiceTest** file.

Although tests are running clean, I'm not sure whether the way I implemented the test in **UpdatePostControllerTest** is the proper way to do it or not. I seek your reviews, thanks in advance.

irem-zeynep commented on practiceapp/src/main/java/com/practiceapp/practiceapp/service/PostService.java:
```diff
@@ -24,9 +25,9 @@
-        return post;
+    public int updatePost(PostEntity postEntity) {
+        postRepository.save(postEntity);
+        return 1;
```
I don't understand why it returns 1. If you don't want to return post entity, maybe you can return a boolean and True or String and "success".

irem-zeynep reviewed (commented):
It's good that you checked both the return status and the response. I just have doubts about the return type as I commented.

schroscatt commented on practiceapp/src/test/java/com/practiceapp/practiceapp/service/PostServiceTest.java:
```diff
@@ -51,4 +51,13 @@ void savePost() {
+        when(postRepository.save(any(PostEntity.class))).thenReturn(postEntity);
+        int response = postService.updatePost(postEntity);
+        assertEquals(response,1);
+
```
Since updatePost function always returns 1, this test is always going to be passed. 
Please consider Zeynep's comment

Ufuk-Arslan commented:
Thanks for your suggestions @irem-zeynep @schroscatt. Here are the changes I implemented with the latest commit:
- Changed the return value of **updatePost** method from integer to **PostEntity** in *UpdatePostController* and *PostService* files.
- Changed the response value controlled in *UpdatePostControllerTest* and *PostServiceTest* files accordinly.
- Included documentation for methods in *UpdatePostController* and *PostService* files.

I tested the code and updating posts works without errors. I hope this is better now.

orhungorkem reviewed (approved):
Comments are comprehensive and tests are selective with the updates offered by Zeynep and Sevde. I think this works!
